custom/icds/translations/integrations/client.py

```diff
@@ -69,7 +69,7 @@ def upload_resource(self, path_to_pofile, resource_slug, resource_name):
         :param resource_name: resource name, mostly same as resource slug itself
         """
         url = "https://www.transifex.com/api/2/project/{}/resources".format(self.project)
-        content = open(path_to_pofile, 'r').read()
+        content = open(path_to_pofile, 'rb').read()
```

Comment:
@nickpell just to confirm that we should mention encoding only when writing?

Reply:
This will not work in Python 3. Just below this data is put in a dict which is then dumped to JSON, but JSON does not support binary data. I believe a better fix is to read this file using the correct encoding. Is it UTF-8 encoded text?

```
Python 3.7.0 (default, Jun 28 2018, 06:01:52)
[Clang 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> content = b"\xe4\x94\x80"
>>> json.dumps({"content": content})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

Reply:
Agree with Daniel.  Also, all uses of ```open``` in text mode should explicitly specify an encoding.